lib/adbc_connection.ex

```diff
@@ -72,6 +72,17 @@ defmodule Adbc.Connection do
     stream_lock(conn, {:query, query, params}, &stream_results/2)
   end
 
+  @doc """
+  Same as `query/3` but raises an exception on error.
+  """
+  @spec query!(t(), binary, [term]) :: result_set
+  def query!(conn, query, params \\ []) when is_binary(query) and is_list(params) do
+    case query(conn, query, params) do
+      {:ok, result} -> result
+      {:error, reason} -> raise "#{inspect(reason)}"
```

Comment:
```suggestion
      {:error, reason} -> raise reason
```

reason is already an exception struct, so we can raise it!